Lesson4.ipynb

```diff
@@ -29,6 +32,655 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Total Steps for each member\n",
+    "\n",
+    "After the Biggest Loser event, I think all the attendees wonder how many steps they clock in total.\n",
+    "\n",
+    "It's not a hard question, let's do coding to get the answer.\n",
+    "\n",
+    "*** Steps statistics ***\n",
+    "\n",
+    "In this part, the premisses we need to know are two things.\n",
+    "\n",
+    "   * Which rows are the data for attendees.\n",
+    "   * Which columns are the data for steps. \n",
+    "\n",
+    "Seen from the table, we can easily figure out the answers.\n",
+    "  \n",
+    "   * -- Rows start from the second row to the end, that is to say, the index starts from 1(We also can say greater than 0, not including 0 -- row_index > 0).\n",
+    "   * -- Columns start from the fifth column to the end, that is to say, the index starts from 4(We also can say greater than 3, not including 3 -- column_index > 3).\n",
+    "   "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Print total steps per person\n",
+    "import csv\n",
+    "with open('Biggest Loser 2018.csv') as csvfile:                                        # Open the CSV file by path\n",
+    "    readCSV = csv.reader(csvfile)                                                      # Read the CSV file\n",
+    "    for row_index,row in enumerate(readCSV):                                           # \"for\" loop statement to fetch the data                     \n",
+    "        if row_index > 0:                                                              # Rows start from 1\n",
+    "            totalSteps = 0                                                             # Define a parameter to store each attendees' total steps\n",
+    "            for column_index, steps_in_column in enumerate(row):                       # Calculate the total steps by using \"for\" loop statement\n",
+    "                if column_index > 3 and steps_in_column != '':                         # Starts from cloumn 4\n",
+    "                    totalSteps = totalSteps + int(steps_in_column)                 \n",
+    "            print(\"Total steps of Member\" + str(row_index) + \" is:\" + str(totalSteps)) # After \"for\" statement completes running, totalSteps is the answer we want"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "From the code block above, we get total steps for every attendee, and print them one by one.\n",
+    "\n",
+    "I think somebody have noticed something, when calculating the total steps, we use a new thing in the code on Line 10\n",
+    "\n",
+    "```python \n",
+    "int(step)\n",
+    "```\n",
+    "** **\n",
+    "\n",
+    "Last week, we know str(), which used for converting other types to string. Now I'm going to talk about int().\n",
```

Comment:
"Last week, we know str(), which used for converting other types to string." should read "Last week, we learnt str(), which is used for converting other types to string."